v1/services/storage/store.go

```diff
@@ -547,10 +547,18 @@ func (s *Store) measurementFields(ctx context.Context, mqAttrs *metaqueryAttribu
 	if err != nil {
 		return nil, err
 	}
-	defer func() { _ = iter.Close() }()
+	defer func() {
+		if iter != nil {
+			_ = iter.Close()
+		}
+	}()
 
 	var fieldNames []string
-	fitr := iter.(query.FloatIterator)
+	fitr, ok := iter.(query.FloatIterator)
+	if !ok {
+		return cursors.NewStringSliceIterator(fieldNames), nil
```

Comment:
In this case, field names will be empty so this will just be an empty StringSliceIterator. Alternatively, we could return an error here that the type assertion failed.

I was unable to reproduce the error so I'm not sure what behavior makes more sense for the conditions under which we might hit this.